### PR TITLE
server: add API to fetch table metadata for table id

### LINKED ISSUE #131222: server: create table metadata api for table id
In addition to the new `GetTableMetadata` api which lists tables per db, we should have an api to retrieve table metadata for a singular table by id.

In addition to the information in system.table_metadata, let's also add the following information to the response:
* CREATE statement for the table - [example](https://github.com/cockroachdb/cockroach/blob/33a791dc1fb7a67b51beebb7bab5f0da8fedc96e/pkg/ui/workspaces/cluster-ui/src/api/tableDetailsApi.ts#L125-L127)
* table grants - ie. [what this query is doing](https://github.com/cockroachdb/cockroach/blob/33a791dc1fb7a67b51beebb7bab5f0da8fedc96e/pkg/ui/workspaces/cluster-ui/src/api/tableDetailsApi.ts#L165-L167) 

Epic: CRDB-37558

Jira issue: CRDB-42450

### PR DESCRIPTION
**Pull request by kyle-a-wong**
Adds a new API v2 endpoint, `GET /api/v2/table_metadata/<id>`, where id is a specific table id.

This API will return table metadata as well as its create statement

Resolves: #131222
Epic: CRDB-37558
Release note: None

### CODE REVIEW

**Review by dhartunian — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/131251)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @kyle-a-wong and @xinhaoz)

___
*[`-- commits` line 2 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/131251#-O7VqxOABYW5N4HHBqVf):*
meganit: "create API" sounds like you're describing a POST. I'd use `add API` here.
___
*[`pkg/server/api_v2_databases_metadata_test.go` line 759 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/131251#-O7VrESfE2R0ZwyJZ6UZ:-O7VrESfE2R0ZwyJZ6U_:b-s92f09) ([raw file](https://github.com/cockroachdb/cockroach/blob/1dee9b598b3eaf92a1489b9656cb02282f083402/pkg/server/api_v2_databases_metadata_test.go#L759)):*
> ```Go
> 	if strings.Contains(contentType, "text/plain") {
> 		data = []byte(fmt.Sprintf(`"%s"`, strings.TrimSpace(string(data))))
> 		fmt.Printf("Data string: %s", data)
> ```

leftover from debugging?
___
*[`pkg/server/api_v2_databases_metadata_test.go` line 762 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/131251#-O7Vr9oP8ioBexzVBzjo:-O7Vr9oP8ioBexzVBzjp:b-ug8yra) ([raw file](https://github.com/cockroachdb/cockroach/blob/1dee9b598b3eaf92a1489b9656cb02282f083402/pkg/server/api_v2_databases_metadata_test.go#L762)):*
> ```Go
> 	}
> 	err = json.Unmarshal(data, &mdResp)
> 	if err != nil {
> ```

This seems redundant with `require.NoError` below. That will print the error if it's present.

**Review by kyle-a-wong — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/131251)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @angles-n-daemons, @dhartunian, and @xinhaoz)

___
*[`-- commits` line 2 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/131251#-O7VqxOABYW5N4HHBqVf):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

meganit: "create API" sounds like you're describing a POST. I'd use `add API` here.
</blockquote></details>

updated commit message
___
*[`pkg/server/api_v2_databases_metadata_test.go` line 759 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/131251#-O7VrESfE2R0ZwyJZ6UZ:-O7_OXC-0h6M5gyaQHuA:ba3dfob) ([raw file](https://github.com/cockroachdb/cockroach/blob/1dee9b598b3eaf92a1489b9656cb02282f083402/pkg/server/api_v2_databases_metadata_test.go#L759)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

leftover from debugging?
</blockquote></details>

Removed
___
*[`pkg/server/api_v2_databases_metadata_test.go` line 762 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/131251#-O7Vr9oP8ioBexzVBzjo:-O7_OY6N64IHo4SmrnRH:ba3dfob) ([raw file](https://github.com/cockroachdb/cockroach/blob/1dee9b598b3eaf92a1489b9656cb02282f083402/pkg/server/api_v2_databases_metadata_test.go#L762)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

This seems redundant with `require.NoError` below. That will print the error if it's present.
</blockquote></details>

Removed

File: pkg/server/api_v2_databases_metadata.go [hunk 1]
```diff
@@ -888,3 +1006,8 @@ type tmJobTriggeredResponse struct {
 	JobTriggered bool             `json:"job_triggered"`
 	Message      JobStatusMessage `json:"message"`
 }
+
+type tableMetadataForIdResponse struct {
```
Comment (xinhaoz) [inline pkg/server/api_v2_databases_metadata.go:1010]: nit: Can we rename this to something like `tableMetadataWithDetails` to signal it's the table metadata with some more information?

File: pkg/server/api_v2_databases_metadata.go [hunk 2]
```diff
@@ -226,6 +230,110 @@ func (a *apiV2Server) GetTableMetadata(w http.ResponseWriter, r *http.Request) {
...
+// - application/json
+//
+// responses:
+//
+//	"200":
+//	  description: A tableMetadataForIdResponse containing the table metadata and table create statement.
+//	"404":
+//		description: If the table for the provided id doesn't exist or the user doesn't have necessary permissions
+//								 to access the table
+func (a *apiV2Server) GetTableMetadataForId(w http.ResponseWriter, r *http.Request) {
```
Comment (xinhaoz) [inline pkg/server/api_v2_databases_metadata.go:256]: Same here with fn naming.

**Review by xinhaoz — APPROVED** [review]
Nice, just a small suggestion on naming.

File: pkg/server/api_v2_databases_metadata.go (hunk 2 above)
Comment (kyle-a-wong) [reply pkg/server/api_v2_databases_metadata.go]: Done

File: pkg/server/api_v2_databases_metadata.go (hunk 1 above)
Comment (kyle-a-wong) [reply pkg/server/api_v2_databases_metadata.go]: Done

**Comment by kyle-a-wong** [comment]
tftr

bors r+

**Pull request merged**